apps/bfd-server/bfd-server-docs/src/main/java/gov/cms/bfd/datadictionary/mappers/CsvToExcel.java

```diff
@@ -0,0 +1,322 @@
+  public static final XSSFColor CUSTOM_BLUE = new XSSFColor(new java.awt.Color(67, 133, 244), null);
+
+  /** Light grey color used for cell borders. */
+  public static final XSSFColor CUSTOM_GREY =
+      new XSSFColor(new java.awt.Color(243, 243, 243), null);
+
+  /** Output stream for writing the Excel workbook file. */
+  private OutputStream outputStream;
+
+  /** Excel data dictionary workbook. */
+  private Workbook workbook;
+
+  /** The current workbook sheet being modified. */
+  private Sheet sheet;
+
+  /** The BFD API version, e.g. V1, V2. */
+  private Version version;
+
+  /**
+   * Creates and instance of the CsvToExcel given and OutputStream, workbook and version.
+   *
+   * @param outputStream the OutputStream to save the Excel workbook to
+   * @param workbook the Excel workbook to load and format
+   * @param version the BFD API version, e.g. V1, V2
+   * @return an CsvToExcel
+   */
+  public static CsvToExcel createInstance(
+      OutputStream outputStream, Workbook workbook, Version version) {
+    var CsvToExcel = new CsvToExcel(outputStream, workbook, version);
+    CsvToExcel.init();
+    return CsvToExcel;
+  }
+
+  /**
+   * Adds a CSV formatted string to the Excel workbook sheet.
+   *
+   * @param csv the CSV formatted String to add
+   */
+  @Override
+  public void accept(String csv) {
+    var record = parseLine(csv);
+    if (record != null) {
+      var rowIndex = sheet.getLastRowNum() + 1;
+      Row row = sheet.createRow(rowIndex);
+      for (int i = 0; i < record.size(); i++) {
+        Cell cell = row.createCell(i);
+        cell.setCellValue(record.get(i));
+      }
+    }
+  }
+
+  /**
+   * Formats and saves the Excel workbook.
+   *
+   * @throws IOException upon write errors
+   */
+  @Override
+  public void close() throws IOException {
+    var widths = ("V1".equalsIgnoreCase(version.name())) ? v1ColumnsWidths : v2ColumnsWidths;
+    formatSheet(widths);
+    workbook.write(outputStream);
+  }
+
+  /**
+   * Private constructor.
+   *
+   * @param outputStream the OutputStream to save the Excel workbook to
+   * @param workbook the Excel workbook to load and format
+   * @param version the BFD API version, e.g. V1, V2
+   */
+  private CsvToExcel(OutputStream outputStream, Workbook workbook, Version version) {
+    this.outputStream = outputStream;
+    this.workbook = workbook;
+    this.version = version;
+  }
+
+  /** Initializes a new Excel workbook sheet for a given BFD API version. */
+  private void init() {
+    sheet = workbook.createSheet(version.name());
+  }
+
+  /**
+   * Applies formatting to data dictionary workbook sheet.
+   *
+   * @param columnsWidths array of column widths to apply to sheet
+   */
+  private void formatSheet(List<Integer> columnsWidths) {
+    var maxRowIndex = sheet.getLastRowNum();
+    var maxColIndex = columnsWidths.size() - 1;
+    searchAndReplace(CellRangeAddress.valueOf("D2:D" + (maxRowIndex + 1)), ";", "\n");
```

Comment:
A nit...might be nice to have some sort of name for the contents "D2:D";  somewhat analogous to XSFSColor variable. Ditto for the "I2:J"